kinto_client/__init__.py

```diff
@@ -199,10 +199,35 @@ def _get_cache_headers(self, safe, data=None, last_modified=None):
             return {'If-Match': utils.quote(last_modified)}
         # else return None
 
+    def _create_if_not_exists(self, resource, **kwargs):
+        try:
+            create_method = getattr(self, 'create_%s' % resource)
+            return create_method(**kwargs)
+        except KintoException as e:
+            if not hasattr(e, 'response') or e.response.status_code != 412:
+                raise e
```

Comment:
Maybe add a comment to specify why using `e.json()['details']['existing']` is not enough